src/palace.pyx

```diff
@@ -660,10 +666,36 @@ cdef class Context:
         If you try to access the resampler list with this index
         without extension, undefined behavior will occur
         (accessing an out of bounds array index).
+
+        This method require the context to be current.
         """
         return self.impl.get_default_resampler_index()
 
-    # TODO: async buffers
+    def precache_buffers_async(self, names: List[str]) -> None:
+        """Prepare asynchronously cached buffers for the given
+        audio files or resource names.
+
+        Duplicate names and buffers already cached are ignored.
+        Cached buffers must be free using `remove_buffer`
+        before destroying the context.
+
+        The `Buffer` objects will be scheduled for loading
+        asynchronously, and should be retrieved later when needed
+        using `buffer_async` or `buffer`.  Buffers that cannot be
+        loaded, for example due to an unsupported format, will be
+        ignored and a later call to `buffer` or `buffer_async` will
+        throw an exception.
+
+        This method require the context to be current.
+        """
+        cdef vector[string] std_names = names
+        cdef vector[alure.StringView] alure_names
+        cdef alure.StringView* alure_name
+        for name in std_names:
+            alure_name = new alure.StringView(name)
+            alure_names.push_back(deref(alure_name))
+            self.precached_buffers.push_back(deref(alure_name))
+        self.impl.precache_buffers_async(alure_names)
```

Comment:
(Continuation of discussion in GH-59)  This does not seem to work (still segfault).  I've been spending the afternoon trying to work it out but haven been able to.  If so won't you before merge, I think it'd be best to leave a FIXME notice above the method.